src/indexer/utils/bitcoin/index.ts

```diff
@@ -0,0 +1,255 @@
+  coinMakerCapService: CoinMarketCapService,
+): Promise<void> {
+  const depositNonce = calculateNonce(blockHeight, txId)
+
+  // Data is in format destinationAddress_destinationDomainID
+  const data = decodedDeposit.data.split("_")
+  const destinationAddress = data[0]
+  const destinationDomainId = parseInt(data[1])
+
+  let amountInUSD
+  try {
+    amountInUSD = await coinMakerCapService.getValueInUSD(decodedDeposit.amount.toString(), decodedDeposit.resource.symbol)
+  } catch (error) {
+    logger.error((error as Error).message)
+    amountInUSD = 0
+  }
+
+  let transfer = await transferRepository.findTransfer(Number(depositNonce), originDomainId, destinationDomainId)
+
+  if (!transfer) {
+    transfer = await transferRepository.insertDepositTransfer({
+      amount: decodedDeposit.amount.toString(),
+      depositNonce: depositNonce,
+      destination: destinationAddress,
+      fromDomainId: originDomainId.toString(),
+      resourceID: decodedDeposit.resource.resourceId,
+      sender: "",
+      toDomainId: destinationDomainId.toString(),
+      usdValue: amountInUSD,
+    })
+  } else {
+    await transferRepository.updateTransfer(
+      {
+        amount: decodedDeposit.amount.toString(),
+        depositNonce: depositNonce,
+        destination: destinationAddress,
+        fromDomainId: originDomainId.toString(),
+        resourceID: decodedDeposit.resource.resourceId,
+        sender: "",
+        toDomainId: destinationDomainId.toString(),
+        usdValue: amountInUSD,
+      },
+      transfer.id,
+    )
+  }
+
+  const deposit = {
+    id: new ObjectId().toString(),
+    type: BitcoinTypeTransfer.Fungible,
+    txHash: txId,
+    blockNumber: blockHeight.toString(),
+    depositData: "",
+    timestamp: new Date(blockTime * 1000),
+    handlerResponse: "",
+    transferId: transfer.id,
+  }
+  await depositRepository.insertDeposit(deposit)
+
+  const feeData = {
+    id: new ObjectId().toString(),
+    transferId: transfer.id,
+    tokenSymbol: decodedDeposit.resource.symbol,
+    decimals: decodedDeposit.resource.decimals,
+    tokenAddress: decodedDeposit.resource.address,
+    amount: decodedDeposit.feeAmount.toString(),
+  }
+  await feeRepository.insertFee(feeData)
+}
+
+async function decodeExecutionEvent(tx: Transaction, resources: BitcoinResource[], client: RPCClient): Promise<DecodedExecution[] | undefined> {
+  let executionData: DecodedExecution[] | undefined
+
+  for (const vout of tx.vout) {
+    if (vout.scriptPubKey.type == OP_RETURN) {
+      // Extract OP_RETURN data (excluding OP_RETURN prefix)
+      const opReturnData = Buffer.from(vout.scriptPubKey.hex, "hex")
+      const data = opReturnData.subarray(2).toString()
+
+      // Data is in format syg_<hash>
+      const dataElems = data.split("_")
+      if (dataElems[0] == "syg") {
+        for (const vin of tx.vin) {
+          const senderTx = (await client.getrawtransaction({ txid: vin.txid, verbose: true })) as Transaction
```

Comment:
Yes, it was updated so we don't have to do it for every tx, now it is only for Sygma executions. Also, it can be easily removed but without it, the code is vulnerable to all kinds of injection or DoS attacks because anyone can put `syg_<url>` in their OP_RETURN making the indexer fetch whatever is there and not actually checking the authenticity of the transaction.

Reply:
Isn't this syg_`IPFS_DATA_HASH` so we woudn't fetch whatever it would just fail if it doesn't exist on IPFS?